src/cmd/run.js

```diff
@@ -272,6 +293,36 @@ export default async function run(
         );
       }
 
+      if (stdin.isTTY && stdin instanceof tty.ReadStream) {
+        readline.emitKeypressEvents(stdin);
+        stdin.setRawMode(true);
+
+        // NOTE: this `Promise.resolve().then(...)` is basically used to spawn a "co-routine" that is executed
+        // before the callback attached to the Promise returned by this function (and it allows the `run` function
+        // to not be stuck in the while loop).
+        Promise.resolve().then(async function() {
+          log.info('Press R to reload (and Ctrl-C to quit)');
+
+          let userExit = false;
```

Comment:
To expand on my comment above about moving this logic to `defaultReloadStrategy`, you're going to need to handle the case where Firefox quits. In other words, try these steps:

* Execute `web-ext run`
* Let Firefox open up
* Go to File > Quit in the Firefox application to close it

After this, `web-ext` should stop running the while loop to check `stdin` for keyboard input but currently it does not. If you move your logic to `defaultReloadStrategy` then you can hook something into the already existing close event:

````js
  firefoxProcess.on('close', () => {
    client.disconnect();
    watcher.close();
    // stop polling stdin...
  });
````